### DISCUSSION TITLE
Searching in the Org Browser

### DISCUSSION THREAD

**Discussion by alexkadis** [forcedotcom/salesforcedx-vscode#5376] [Ideas]
**Is your feature request related to a problem? Please describe.**
The Org Browser has dozens of different types of metadata, and there may be hundreds of components or classes within each metadata type.

**Describe the solution you'd like**
I'd like a search box at the top includes both metadata types and file names. This would be enabled after refreshing types.

**Example of what it could look like:**
![search-org-metadata](https://user-images.githubusercontent.com/4106499/74964883-37141a00-53e2-11ea-882f-6fb168bd140c.png)

**Comment by lcampos** [comment]
Hi @alexkadis , thanks for opening this issue. Currently there are some limitation with VSCode's UI, for example we can't add an input in order to do search, etc. What is currently available is to start highlighting text that matches what you type if you've previously focused the Org Browser. If you look at the gif below, I start typing "Visualforce" and the focus of the Org Browser goes to the items that match that string.

![search-org-browser](https://user-images.githubusercontent.com/1041842/75183079-9893fb80-56f6-11ea-95c9-efad6270ada4.gif)

**Comment by AllanOricil** [comment]
@lcampos you can use the "webview view" (weird name, but that is how they called it. And it is not the same as webview) api to implement a custom Explorer. In this Explorer you will render the same list but you have control of the entire UI, which means you will be able to add a search bar.
I dont think vscode will add a search bar in the standard Explorer, so that is why I'm suggesting to your team to use this other API to implement this feature.

**Comment by lcampos** [comment]
hi @AllanOricil I'm no longer a maintainer on this project but @smaddox-sf and the rest of the team can help look into this. I do agree with you that vscode will probably not add a search bar in the near future.

**Comment by AllanOricil** [comment]
I still believe this has to be implemented with a webview view

**Comment by Sam-Cousins** [comment]
@smaddox-sf 

Any ideas on whether we can get this? It's so painful at the moment. Some third party extensions have this ability, but it'd be nice for official SF support!
